File: services/credit-service/src/main/java/com/bankapp/creditservice/dto/CreditCalculationResponse.java
package com.bankapp.creditservice.dto;

import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreditCalculationResponse {

    private BigDecimal monthlyPayment;
    private BigDecimal interestRate;
    private BigDecimal approvedLimit;
    private String status;
}

File: mobile/shared/src/commonMain/kotlin/com/bankapp/zbk/shared/dto/CreditCalculationResponse.kt
package com.bankapp.zbk.shared.dto

import kotlinx.serialization.Serializable

/**
 * Mirrors OpenAPI `CreditCalculationResponse`.
 * Credit-service JSON has no `path` (or other gateway) field — keep this DTO strict.
 */
@Serializable
data class CreditCalculationResponse(
    val monthlyPayment: Double,
    val interestRate: Double,
    val approvedLimit: Double,
    val status: String,
)
